code/modules/atmospherics/machinery.dm

```diff
@@ -1,62 +1,52 @@
-/*
-Quick overview:
-
-Pipes combine to form pipelines
-Pipelines and other atmospheric objects combine to form pipe_networks
-	Note: A single pipe_network represents a completely open space
-
-Pipes -> Pipelines
-Pipelines + Other Objects -> Pipe network
-
+ABSTRACT_TYPE(/obj/machinery/atmospherics)
+
+/**
+ * Quick overview:
+ * Pipes combine to form pipelines.
+ * Pipelines and other atmospheric objects combine to form pipe_networks.
+ * Note: A single pipe_network represents a completely open space.
+ *
+ * Pipes -> Pipelines
+ * Pipelines + Other Objects -> Pipe network
 */
-//
-obj/machinery/atmospherics
+/obj/machinery/atmospherics
 	anchored = ANCHORED
 
 	var/initialize_directions = 0
 
-	process()
-		build_network()
-		..()
-
-	// override default subscribes to be in a different process loop. that's why they don't call parent ( ..() )
-	SubscribeToProcess()
-		START_TRACKING_CAT(TR_CAT_ATMOS_MACHINES)
-
-	UnsubscribeProcess()
-		STOP_TRACKING_CAT(TR_CAT_ATMOS_MACHINES)
-
-	proc
-		network_disposing(datum/pipe_network/reference)
-			// Called by a network associated with this machine when it is being disposed
-			// This must be implemented to unhook any references to the network
-
-			return null
+/obj/machinery/atmospherics/process()
+	build_network()
+	..()
 
-		network_expand(datum/pipe_network/new_network, obj/machinery/atmospherics/pipe/reference)
-			// Check to see if should be added to network. Add self if so and adjust variables appropriately.
-			// Note don't forget to have neighbors look as well!
+// override default subscribes to be in a different process loop. that's why they don't call parent ( ..() )
+/obj/machinery/atmospherics/SubscribeToProcess()
+	START_TRACKING_CAT(TR_CAT_ATMOS_MACHINES)
 
-			return null
+/obj/machinery/atmospherics/UnsubscribeProcess()
+	STOP_TRACKING_CAT(TR_CAT_ATMOS_MACHINES)
 
-		build_network()
-			// Called to build a network from this node
+/// Called by a network associated with this machine when it is being disposed.
+/// This must be implemented to unhook any references to the network.
+/obj/machinery/atmospherics/proc/network_disposing(datum/pipe_network/reference)
 
-			return null
+/// Check to see if should be added to network. Add self if so and adjust variables appropriately.
+/// Note don't forget to have neighbors look as well!
+/obj/machinery/atmospherics/proc/network_expand(datum/pipe_network/new_network, obj/machinery/atmospherics/pipe/reference)
 
-		return_network(obj/machinery/atmospherics/reference)
-			// Returns pipe_network associated with connection to reference
-			// Notes: should create network if necessary
-			// Should never return null
+/// Called to build a network from this node.
+/obj/machinery/atmospherics/proc/build_network()
```

Comment:
wait so is there just no code in this proc?

Reply:
its a parent proc meant to be overriden

Reply:
i see